controllers/listingController.js

```diff
@@ -79,39 +75,70 @@ var ListingController = {
 
     createListing : (item) => {
 
-        console.log('CREATING LISTING WITH ITEM:');
-        console.log(item);
+        console.log('CREATING LISTING');
+        let variants = [];
+        item.variants.forEach( v => {
+            var variant = {
+                vendorId: v.id,
+                title: v.title,
+                sku: v.sku,
+                option1: v.option1,
+                option2: v.option2,
+                option3: v.option3,
+                imageId: v.image_id,
+                inventoryQuantity: v.inventory_quantity
+            };
+            variants.push(variant);
+        });
+
         var listing = new Listing({
             title: item.title,
             description: item.body_html,
             index: 0,
             isActive: true,
             vendorId: item.id,
-            previewImageUrl: item.images[0].src,
+            feedImageUrl: item.images[0].src,
             lastUpdatedAt: item.updated_at,
-            price: item.variants[0].price
+            price: item.variants[0].price,
+            variants: variants
         });
 
         ListingController.saveListing(listing);
     },
 
     updateListing : (listing, item) => {
+        let variants = [];
+        item.variants.forEach( v => {
+            var variant = {
+                vendorId: v.id,
+                title: v.title,
+                sku: v.sku,
+                option1: v.option1,
+                option2: v.option2,
+                option3: v.option3,
+                imageId: v.image_id,
+                inventoryQuantity: v.inventory_quantity
+            };
+            variants.push(variant);
+        });
+
         listing.title = item.title;
         listing.description = item.body_html;
-        listing.previewImageUrl = item.images[0].src;
+        listing.feedImageUrl = item.images[0].src;
```

Comment:
There's got to be something that prevents this from happening. Some kind of compilation-like check that doesn't let you run with a reference to a value/function that doesn't exist.

Reply:
I think there is. Mongoose has some kind of .valid check we could run I think.